code/modules/client/client_procs.dm

```diff
@@ -110,6 +110,18 @@
 
 	..()	//redirect to hsrc.Topic()
 
+/client/proc/generate_clickcatcher()
```

Comment:
i dont see a need to seperate generate and apply here, we wouldn't want a click catcher that isn't sized properly, right?

Reply:
> i dont see a need to seperate generate and apply here, we wouldn't want a click catcher that isn't sized properly, right?

It works fine thanks to the Upgradegreed.